internal/profile/profile.go

```diff
@@ -436,6 +440,30 @@ func EnhanceFromSchematic(
 				return prof, fmt.Errorf("error getting extension image %s: %w", overlayRef.TaggedReference, err)
 			}
 
+			overlayProfilePath, err := artifactProducer.GetOverlayArtifact(ctx, artifacts.Arch(prof.Arch), overlayRef, artifacts.OverlayKindProfiles)
+			if err != nil {
+				return prof, fmt.Errorf("error getting overlay profiles %s: %w", overlayRef.TaggedReference, err)
+			}
+
+			var overlayProfile profile.Profile
+
+			overlayProfileData, err := os.ReadFile(filepath.Join(overlayProfilePath, overlayRef.Name+".yaml"))
+			if err != nil {
+				return prof, fmt.Errorf("error reading overlay profile %s: %w", overlayProfilePath, err)
+			}
+
+			if err := yaml.Unmarshal(overlayProfileData, &overlayProfile); err != nil {
+				return prof, fmt.Errorf("error unmarshalling overlay profile %s: %w", overlayProfilePath, err)
+			}
+
+			if overlayProfile.Output.ImageOptions != nil {
```

Comment:
that is the worst part - I basically just copy the bootloader option here, but I don't know what can be taken and what not, or whether we should instead flip the stuff around and start with the initial base profile and apply Image Factory options on top of it?

Reply:
reworked this part completely